connector/json-test-connector/src/main.rs

```diff
@@ -1,13 +1,14 @@
 mod source;
 
 use fluvio::{TopicProducer, RecordKey};
-use fluvio_connector_common::{Source, connector, ConnectorConfig, Result};
+use fluvio_connector_common::{Source, connector, Result};
 use futures::StreamExt;
+use serde::Deserialize;
 
 use crate::source::TestJsonSource;
 
 #[connector(source)]
-async fn start(config: ConnectorConfig, producer: TopicProducer) -> Result<()> {
+async fn start(config: CustomConfig, producer: TopicProducer) -> Result<()> {
```

Comment:
Why renaming?

Reply:
This is a demonstration that the user can have its own type instead of our `ConnectorConfig` (if he wants). It gives the ability to have a custom configuration in free format in the `config.yaml` file. The config file still has to be valid against `ConnectorConfig` type, we parse and use it. However, in the custom config type, the user can use an only subset of  `ConnectorConfig`, or have properties that are not present in the `ConnectorConfig` at all.

Right now, the only way to customize connector configuration is using `parameters` section. This is a very limited and inconvenient approach, which always ends up with manual parsing. Inviting @simlay to the thread to share if it is really painful for connector developers.

This also helps with code decoupling: `ConnectorConfig` from SDK will mostly be used by code from SDK (init Producer/Consumer), and custom user-defined config will be living within the codebase of the connector and will evolve along.